src/core/createLifecycle.js

```diff
@@ -70,25 +69,14 @@ export default componentRegistry => {
     onComponentsRegistered: tools => {
       return invokeHook(componentRegistry, "onComponentsRegistered", tools);
     },
-    onBeforeViewStart: guardLifecycleMethod(payload => {
-      return invokeHook(componentRegistry, "onBeforeViewStart", payload);
+    onBeforeEvent: guardLifecycleMethod((event, isViewStart) => {
+      return invokeHook(componentRegistry, "onBeforeEvent", event, isViewStart);
     }),
-    onViewStartResponse: guardLifecycleMethod(response => {
-      return invokeHook(
-        componentRegistry,
-        "onViewStartResponse",
-        createResponse(response)
-      );
+    onBeforeRequest: guardLifecycleMethod(payload => {
+      return invokeHook(componentRegistry, "onBeforeRequest", payload);
     }),
-    onBeforeEvent: guardLifecycleMethod(payload => {
-      return invokeHook(componentRegistry, "onBeforeEvent", payload);
-    }),
-    onEventResponse: guardLifecycleMethod(response => {
-      return invokeHook(
-        componentRegistry,
-        "onEventResponse",
-        createResponse(response)
-      );
+    onResponse: guardLifecycleMethod(response => {
+      return invokeHook(componentRegistry, "onResponse", response);
```

Comment:
I brought these changes up over Slack. The gateway has changed so that multiple events can be sent on a single request. Right now, we’re always sending one event per request, but I suspect this will change, and now that things like the `web` context must be applied to the event and not on the top-level of the payload, we needed to make some architectural changes. Previously we had:

`onBeforeViewStart`
`onBeforeEvent`
`onViewStartResponse`
`onEventResponse`

And this PR changes those to:

`onBeforeEvent`
`onBeforeRequest`
`onResponse`